packages/documentation/copy/en/reference/Variable Declarations.md

```diff
@@ -704,3 +704,108 @@ clone.m(); // error!
 
 Second, the TypeScript compiler doesn't allow spreads of type parameters from generic functions.
 That feature is expected in future versions of the language.
+
+## `using` declarations
+
+`using` declarations are an upcoming feature for JavaScript that are part of the 
+[Stage 3 Explicit Resource Management](https://github.com/tc39/proposal-explicit-resource-management) proposal. A `using`
+declaration is much like a `const` declaration, except that it couples the _lifetime_ of the value bound to the declaration with
+the _scope_ of the variable.
+
+When control exits the block containing a `using` declaration, the `[Symbol.dispose]()` method of the
+declared value is executed, which allows that value to perform cleanup:
+
+```ts
+function f() {
+  using x = new C();
+  doSomethingWith(x);
+} // `x[Symbol.dispose]()` is called
+```
+
+At runtime, this has an affect _roughly_ similar to the following:
+
+```ts
+function f() {
+  const x = new C();
+  try {
+    doSomethingWith(x);
+  }
+  finally {
+    x[Symbol.dispose]();
+  }
+}
+```
+
+It's important to note that the value can be `null` or `undefined`, in which case nothing is disposed at the end of the block:
+
+```ts
+{
+  using x = b ? new C() : null;
+}
+```
+
+`using` declarations are extremely useful for avoiding memory leaks when working with JavaScript object's that hold on to native
+references like file handles, or scoped operations like tracing:
+
+```ts
+{
+  using file = await openFile();
+  file.write(text);
+  doSomethingThatMayThrow();
+} // `file` is disposed, even if an error is thrown
+```
+
+Unlike `var`, `let`, and `const`, `using` declarations do not support destructuring.
+
+## `await using` declarations
+
+Some resources or operations may have cleanup that needs to be performed asynchronously. To accommodate this, the
+[Explicit Resource Management](https://github.com/tc39/proposal-explicit-resource-management) proposal also introduces the
+`await using` declaration:
+
+```ts
+async function f() {
+  await using x = new C();
+} // `await x[Symbol.asyncDispose]()` is invoked
+```
+
+An `await using` declaration invokes, and _awaits_, its value's `[Symbol.asyncDispose]()` method as control leaves the containing
+block. This allows for asynchronous cleanup, such as a database transaction performing a rollback or commit.
```

Comment:
Should there be an example with an async right side, just to help dismambiguate that `await using x = foo()` does not await the result of `foo`?

Reply:
I can look into expanding this. I'm not sure how much detail we want to add here. I could also cover how `using` and `await using` can be used in `for` and `for..of`, but worried that might be too much in the weeds.